pyg_lib/csrc/ops/cpu/index_sort_kernel.cpp

```diff
@@ -0,0 +1,85 @@
+#include <tuple>
+
+#include <ATen/ATen.h>
+#include <ATen/Parallel.h>
+#include <torch/library.h>
+
+#include "radix_sort.h"
+
+namespace pyg {
+namespace ops {
+
+namespace {
+
+template <typename scalar_t>
+void vectorized_copy(scalar_t* dst, const scalar_t* src, int64_t size) {
+  constexpr int64_t unfold_step = 4;
+  int64_t index;
+  if (size >= unfold_step) {
+#pragma omp simd
+    for (index = 0; index < size; index += unfold_step) {
+      dst[index] = src[index];
+      dst[index + 1] = src[index + 1];
+      dst[index + 2] = src[index + 2];
+      dst[index + 3] = src[index + 3];
+    }
+  }
+  for (index; index < size; ++index) {
+    dst[index] = src[index];
+  }
+}
+
+std::tuple<at::Tensor, at::Tensor> index_sort_kernel(
+    const at::Tensor& input,
+    const at::optional<int64_t> max) {
+  TORCH_CHECK(input.is_contiguous(), "Input should be contiguous.")
+  TORCH_CHECK(input.dim() == 1, "Input should be 1-dimensional.");
+  if (input.numel() > at::internal::GRAIN_SIZE && is_radix_sort_available()) {
+    const auto elements = input.numel();
+    const auto maximum = max.value_or(at::max(input).item<int64_t>());
```

Comment:
```suggestion
    const auto maximum = max.value_or(input.max().item<int64_t>());
```
Cool, learned something :)